### PR TITLE
feat: Add S3OutputPath to EndpointConfig in SDK config defaults

### PR DESCRIPTION
**Pull request by Captainia**
*Issue #, if available:* 

*Description of changes:* This change adds the `S3OutputPath` from the [EndpointConfig.AsyncInferenceConfig](https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_AsyncInferenceOutputConfig.html#sagemaker-Type-AsyncInferenceOutputConfig-S3OutputPath) to the intelligent defaults.

*Testing done:* Unittests

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/unit/sagemaker/config/test_config_schema.py [hunk 1]
```diff
@@ -199,3 +199,13 @@ def test_invalid_s3uri_schema(base_config_with_schema):
     config["SageMaker"] = {"PythonSDK": {"Modules": {"RemoteFunction": {"S3RootUri": "bad_regex"}}}}
     with pytest.raises(exceptions.ValidationError):
         validate(config, SAGEMAKER_PYTHON_SDK_CONFIG_SCHEMA)
+
+
+def test_invalid_endpoint_config_s3uri_schema(base_config_with_schema):
```
Comment (evakravi) [inline tests/unit/sagemaker/config/test_config_schema.py:204]: can we add more unit tests?

File: src/sagemaker/model.py [hunk 2]
```diff
@@ -1301,6 +1302,11 @@ def deploy(
                 ENDPOINT_CONFIG_ASYNC_KMS_KEY_ID_PATH,
                 sagemaker_session=self.sagemaker_session,
             )
+            async_inference_config.output_path = resolve_value_from_config(
```
Comment (evakravi) [inline src/sagemaker/model.py:1305]: will this only affect async inference? Can you add tests to confirm the intelligent default value gets passed?

File: src/sagemaker/config/config_schema.py [hunk 3]
```diff
@@ -135,6 +136,9 @@ def _simple_path(*args: str):
 ENDPOINT_CONFIG_ASYNC_KMS_KEY_ID_PATH = _simple_path(
```
Comment (rubanh) [inline src/sagemaker/config/config_schema.py:136]: config injection that uses ENDPOINT_CONFIG_ASYNC_INFERENCE_PATH will also plug in this new value. Can we add or update unit tests for model.py and for session.py? Existing unit tests should have `_with_sagemaker_config_injection` in the name. Might be able to just add to the existing ones.

File: tests/unit/sagemaker/config/test_config_schema.py [hunk 4]
```diff
@@ -199,3 +199,13 @@ def test_invalid_s3uri_schema(base_config_with_schema):
     config["SageMaker"] = {"PythonSDK": {"Modules": {"RemoteFunction": {"S3RootUri": "bad_regex"}}}}
     with pytest.raises(exceptions.ValidationError):
         validate(config, SAGEMAKER_PYTHON_SDK_CONFIG_SCHEMA)
+
```
Comment (rubanh) [inline tests/unit/sagemaker/config/test_config_schema.py:202]: would be great to also include this new parameter in tests/data/config/config.yaml, and in test_valid_endpoint_config_schema

**Review by rubanh — CHANGES_REQUESTED** [review]

File: src/sagemaker/model.py (hunk 2 above)
Comment (JGuinegagne) [reply src/sagemaker/model.py:1305]: +1

**Review by JGuinegagne — CHANGES_REQUESTED** [review]
Could you please also update the [documentation](https://sagemaker.readthedocs.io/en/stable/overview.html#configuring-and-using-defaults-with-the-sagemaker-python-sdk) as well?

I think you need to add the extra attribute [here](https://github.com/aws/sagemaker-python-sdk/blob/bfbb35feb5577d14446a6a06589b08f80aef4cbc/doc/overview.rst?plain=1#L1871)

**Comment by knikure** [comment]
@Captainia Are you still working on this PR? Can you resolve conflicts and address comments?

**Pull request closed without merging**